backends/v3/src/queue.rs

```diff
@@ -364,7 +364,7 @@ impl State {
                         // Add it back to the front
                         tracing::debug!("Over budget: not enough free blocks");
                         self.entries.push_front((id, entry));
-                        break;
+                        continue;
```

Comment:
Important so we either allocate the rest or push them back into entries (otherwise they get discarded without being run)

Reply:
I missed the PR where this was moved.
Since we check the prefill token budget early, if all entries in `batch`end up being in the radix tree, can't we end up running at very low prefill utilization then? 
If I understand all the components correctly, the block allocator is the only one to truly know the length in tokens that each request will have.